### PR TITLE
Overridden methods with wider access don't have it applied in specialized versions

### PR DESCRIPTION
**Pull request by jxnu-liguobin**
When the method is public, the specialized method should be public in override.

scala/bug#10211

### CODE REVIEW

File: src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala [hunk 1]
```diff
@@ -1123,7 +1125,7 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
             }
           } else None
         case (overridden, env) =>
-          val om = specializedOverload(clazz, overridden, env)
+          val om = specializedOverload(clazz, overridden, env, excludeFlags = overriding.isPublic)
```
Comment (lrytz) [inline src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala]: Would this work instead, without changing the method `specializedOverload`?
```suggestion
          val om = specializedOverload(clazz, overriding, env)
```

Comment (jxnu-liguobin) [reply src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala]: Yes, I've tried, but failed, because there is still exists `protected local`

**Comment by SethTisue** [comment]
@jxnu-liguobin do you believe this is ready for review? I see that CI likes it.

@lrytz your question was just a side question and not a blocker, I think? does @jxnu-liguobin's answer satisfy you?

**Comment by lrytz** [comment]
It seemed to me that my suggestion could be the more principled solution, so I should spend some time finding out why it doesn't work.

**Comment by SethTisue** [comment]
@lrytz do you intend to return to this?

**Comment by SethTisue** [comment]
ping @lrytz

**Review by lrytz — APPROVED** [review]

**Comment by SethTisue** [comment]
thank you @jxnu-liguobin !

**Pull request merged**